### ISSUE TITLE
Operator [] for string

### ISSUE THREAD

**Issue by rhysd** [rhysd/Dachs#35]
Value of `string` is now only a constant.  Size of allocated memory for constant can be known and it means that conversion from constant `i8**` to `[n x i8]` is possible.  Though `string` is a null-terminated string, last `\0` is not be included in the result of the cast.

When this issue is closed successfully, index access to string is available by casting from `string` to `[char]`.

**Comment by rhysd** [comment]
I've noticed that it's impossible.

```
func main
  var s := "aaa"
  if __builtin_read_cycle_counter() % 2u == 0u
    s = "b"
  end

  buf := s as [char] # length of s can't be determined!
end
```

As alternative, operator `[]` for `string` should be implemented to archive the target of this issue.

**Comment by rhysd** [comment]
Implemented in eba33b55c2fc2927634cfb63091d5f132c2c9c1e
